modules/core/src/system.cpp

```diff
@@ -1489,28 +1491,51 @@ void TlsAbstraction::setData_(void *pData)
 }
 #endif // WINRT
 #else // _WIN32
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
 static void opencv_tls_destructor(void* pData);
+#endif // OPENCV_DISABLE_THREAD_SUPPORT
 TlsAbstraction::TlsAbstraction()
 {
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
     CV_Assert(pthread_key_create(&tlsKey, opencv_tls_destructor) == 0);
+#else // OPENCV_DISABLE_THREAD_SUPPORT
+    CV_Error(cv::Error::StsNotImplemented,
+             "TlsAbstraction is disabled by OPENCV_DISABLE_THREAD_SUPPORT=ON");
+#endif // OPENCV_DISABLE_THREAD_SUPPORT
 }
 TlsAbstraction::~TlsAbstraction()
 {
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
     cv::__termination = true;  // DllMain is missing in static builds
     if (pthread_key_delete(tlsKey) != 0)
     {
         // Don't use logging here
         fprintf(stderr, "OpenCV ERROR: TlsAbstraction::~TlsAbstraction(): pthread_key_delete() call failed\n");
         fflush(stderr);
     }
+#else // OPENCV_DISABLE_THREAD_SUPPORT
+    CV_Error(cv::Error::StsNotImplemented,
+             "TlsAbstraction is disabled by OPENCV_DISABLE_THREAD_SUPPORT=ON");
```

Comment:
destructors must not throw exceptions.

Introduced changes about TLS should be minimized.
Go on upper level, not need to "stub" local functions - work with public API only.

Reply:
> destructors must not throw exceptions.
> 
Sure! Why? (Genuine question, I am not challenging your comment!)

> Introduced changes about TLS should be minimized.
> Go on upper level, not need to "stub" local functions - work with public API only.

What does "go on upper level" mean? Leave `TlsAbstraction` unchanges, and use the `#ifundef / #else / #endif` in all places where `TlsAbstraction` is used?

Reply:
Main goal here is to minimize amount of `#if` conditions.

Public API for TLS is `TLSDataContainer`. Other things (like `TlsAbstraction`, `TlsStorage`) are local implementation details - we should not care about them here.
Stub `TLSDataContainer` with straightforward implementation.

Reply:
Done.